symforce/slam/imu_preintegration/manifold_symbolic.py

```diff
@@ -0,0 +1,349 @@
+    """
+    Calculates the derivatives of the new state (meaning the DR, Dv, and Dp) w.r.t. the previous state,
+    gyroscope measurement, and the accelerometer mesaurement.
+
+
+    Args:
+        DR (sf.Rot3): Preintegrated DR of the previous state
+        corrected_gyro (sf.V3): gyroscope measurment corrected for IMU bias
+        corrected_accel (sf.V3): accelerometer measurement corrected for IMU bias
+        dt (T.Scalar): Time difference between the previous time step and the new time step
+        epsilon (T.Scalar): epsilon for numerical stability
+
+    Returns:
+        T.Tuple[sf.M99, sf.M93, sf.M93]: new_state_D_old_state, new_state_D_gyro_measurement, new_state_D_accel_measurement
+    """
+    # calculate new_state_D_old_state
+    DR_update = sf.Rot3.from_tangent(corrected_gyro * dt, epsilon)
+    new_DR_D_old_DR = DR_update.inverse().to_rotation_matrix()
+    new_DR_D_old_Dv = sf.M33.zero()
+    new_DR_D_old_Dp = sf.M33.zero()
+
+    new_Dv_D_old_DR = -DR.to_rotation_matrix() * sf.Matrix.skew_symmetric(corrected_accel * dt)
+    new_Dv_D_old_Dv = sf.M33.eye()
+    new_Dv_D_old_Dp = sf.M33.zero()
+
+    new_Dp_D_old_DR = (
+        (-DR.to_rotation_matrix() * sf.Matrix.skew_symmetric(corrected_accel * dt)) * dt / 2
+    )
+    new_Dp_D_old_Dv = dt * sf.M33.eye()
+    new_Dp_D_old_Dp = sf.M33.eye()
+
+    new_state_D_old_state = sf.Matrix.block_matrix(
+        [
+            [new_DR_D_old_DR, new_DR_D_old_Dv, new_DR_D_old_Dp],
+            [new_Dv_D_old_DR, new_Dv_D_old_Dv, new_Dv_D_old_Dp],
+            [new_Dp_D_old_DR, new_Dp_D_old_Dv, new_Dp_D_old_Dp],
+        ]
+    )
+
+    # calculate new_D_gyro
+    new_DR_D_gyro = right_jacobian(corrected_gyro * dt, epsilon) * dt
+    new_DvDp_D_gyro = sf.M63.zero()
+
+    new_state_D_gyro = sf.Matrix.block_matrix(
+        [
+            [new_DR_D_gyro],
+            [new_DvDp_D_gyro],
+        ]
+    )
+
+    # calculate new_D_accel
+    new_DR_D_accel = sf.M33.zero()
+    new_Dv_D_accel = DR.to_rotation_matrix() * dt
+    new_Dp_D_accel = DR.to_rotation_matrix() * (dt * dt / 2)
+
+    new_state_D_accel = sf.Matrix.block_matrix(
+        [
+            [new_DR_D_accel],
+            [new_Dv_D_accel],
+            [new_Dp_D_accel],
+        ]
+    )
+
+    return sf.M99(new_state_D_old_state), sf.M93(new_state_D_gyro), sf.M93(new_state_D_accel)
+
+
+def imu_manifold_preintegration_update(
+    # Initial state
+    DR: sf.Rot3,
+    Dv: sf.V3,
+    Dp: sf.V3,
+    covariance: sf.M99,
+    DR_D_gyro_bias: sf.M33,
+    Dv_D_accel_bias: sf.M33,
+    Dv_D_gyro_bias: sf.M33,
+    Dp_D_accel_bias: sf.M33,
+    Dp_D_gyro_bias: sf.M33,
+    # Biases and noise model
+    accel_bias: sf.V3,
+    gyro_bias: sf.V3,
+    accel_cov_diagonal: sf.V3,
+    gyro_cov_diagonal: sf.V3,
+    # Measurement
+    accel_measurement: sf.V3,
+    gyro_measurement: sf.V3,
+    dt: T.Scalar,
+    # Singularity handling,
+    epsilon: T.Scalar,
+    # Configuration
+    use_handwritten_derivatives: bool = True,
```

Comment:
Don't we want to default this to false?

Reply:
Well, I was thinking that it made sense to use the handwritten version because the two versions should be the same (in terms of correctness), but the handwritten version is faster. So, as a signaling mechanism as to which version you should use, I thought it made sense to default to the faster one.

But, now that I think about it, there's really no need to give a default value. I don't think there's any harm though for me to independently want to change it.

Reply:
I somehow read this option as the inverse, I thought this was defaulting to automatic derivatives.  We should default to using the handwritten derivatives like this already is since that's what we think you should use